include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -917,8 +917,8 @@ namespace VectorTools
       additional_data.mapping_update_flags = (update_values | update_JxW_values);
       MatrixFree<dim, Number> matrix_free;
       matrix_free.reinit (mapping, dof, constraints,
-                          QGauss<1>(fe_degree+1), additional_data);
-      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, 1, Number> MatrixType;
+                          QGauss<1>(fe_degree+2), additional_data);
+      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, fe_degree+2, 1, Number> MatrixType;
```

Comment:
Generally I do not expect projection to be a bottleneck in any FEM code. Most likely it is a pre-processing or post-processing step. So `fe_degree+2` vs `fe_degree+1` would probably not matter. On another hand, @masterleinad commented on previously implemented `project()` with SparseMatrix that we used `fe_degree+2` to assemble the mass matrix. My interpretation is that as soon as you have distorted elements, `fe_degree+1` would make the mass matrix a bit under-integrated. So `fe_degree+2` is a safer bet.

My comment was addressed not to this function, but the one below, where I want to use `project()` from MatrixFree data with `fe_degree+1` quadrature points (say when I solve Laplace equation and want to project some quadrature point data to FE space).